### PR TITLE
SONARJAVA-2617 Handle jetbrains nullness annotations

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java [hunk 1]
```diff
@@ -357,32 +363,34 @@ public void addExceptionalYield(SymbolicValue target, ProgramState exceptionalSt
     return stateStream.collect(Collectors.toList());
   }
 
-  private static boolean parametersAreNullable(MethodTree methodTree) {
-    return isAnnotatedWith(methodTree, "javax.annotation.ParametersAreNullableByDefault");
+  private static boolean isAnnotatedWith(MethodTree methodTree, String annotation) {
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java]: This one probably deserve a better name, as it his checking in enclosing class and package.

File: java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java [hunk 2]
```diff
@@ -350,39 +349,26 @@ public void addExceptionalYield(SymbolicValue target, ProgramState exceptionalSt
...
 
-  private static boolean parametersAreNullable(MethodTree methodTree) {
-    return isAnnotatedWith(methodTree, "javax.annotation.ParametersAreNullableByDefault");
-  }
-
-  private static boolean parametersAreNonNull(MethodTree methodTree) {
-    return isAnnotatedWith(methodTree, "javax.annotation.ParametersAreNonnullByDefault");
-  }
-
   private static boolean isAnnotatedWith(MethodTree methodTree, String annotation) {
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java]: I think this one should be also refactored to AnnotationUtils.

File: java-frontend/src/main/java/org/sonar/java/se/AnnotationUtils.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+import org.sonar.plugins.java.api.semantic.Symbol;
+
+public final class AnnotationUtils {
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/AnnotationUtils.java]: I tend to think we may want to refine a bit the name : NullableAnnotationUtils ?

File: java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java [hunk 4]
```diff
@@ -721,7 +707,7 @@ private void enqueueHappyPath(ProgramState programState, MethodInvocationTree mi
   }
 
   private ProgramState handleSpecialMethods(ProgramState ps, MethodInvocationTree mit) {
-    if (isNonNullMethod(mit.symbol())) {
+    if (AnnotationUtils.isAnnotatedNonNull(mit.symbol())) {
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java]: Wondering about having a static import for AnnotationUtils method.

**Pull request merged**